shared/pkg/families/manager.go

```diff
@@ -77,17 +78,43 @@ func New(logger *log.Entry, config *Config) *Manager {
 
 type RunErrors map[types.FamilyType]error
 
-func (m *Manager) Run() (*results.Results, RunErrors) {
-	familyErrors := make(map[types.FamilyType]error)
+type familyResult struct {
+	result interfaces.IsResults
+	err    error
+}
 
+func (m *Manager) Run(ctx context.Context) (*results.Results, RunErrors) {
+	familyErrors := make(RunErrors)
 	familyResults := results.New()
+	result := make(chan familyResult, len(m.families))
 
+outer:
 	for _, family := range m.families {
-		ret, err := family.Run(familyResults)
-		if err != nil {
-			familyErrors[family.GetType()] = fmt.Errorf("failed to run family %v: %w", family.GetType(), err)
-		} else {
-			familyResults.SetResults(ret)
+		go func() {
+			ret, err := family.Run(familyResults)
+			result <- familyResult{
+				ret,
+				err,
+			}
+		}()
+
+	inner:
+		for {
+			select {
+			case <-ctx.Done():
+				log.Warningf("received context cancelled while family %q is still running. Aborting...", family)
```

Comment:
I think if a particular family is aborted while its still running it should be counted as a family error e.g.

```
familyErrors[family.GetType()] = fmt.Errorf("failed to run family %v: aborted", family.GetType(), r.err)
```

Reply:
Fixed.